### PR TITLE
initial commit for factorial

### PR DESCRIPTION
**Pull request by lather10**

### CODE REVIEW

File: factorial/fact.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    else
+    {
+            n=n*fun(n-1);
+    }
+}
+int main()
+{
+    int n;
+    char c;
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: What is the usage of this character variable?

File: factorial/fact.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    std::cout<<"Enter a positive Integer to find Factorial: ";
+    std::cin>>n;
+    std::cout<<"\n";
+    if(n>=0 && n!=c)
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: Why are you comparing integer n to an uninitialized char variable c.
This is undefined behaviour and might crash your program

File: factorial/fact.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#include<iostream>
+int fun(int n)
+{
+if(n==0)
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: Indentation looks off

File: factorial/fact.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        return 1;
+    }
+    else
+    {
+            n=n*fun(n-1);
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: Shouldn't we return n*fun(n-1)

File: factorial/fact.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+}
+int main()
+{
+    int n;
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: What is n? Try to give meaningful names to variables. If this variable is supposed to hold input then you can name it as input or userInput. Try to read C++ naming convention from google

File: factorial/fact.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    char c;
+    std::cout<<"Enter a positive Integer to find Factorial: ";
+    std::cin>>n;
+    std::cout<<"\n";
+    if(n>=0 && n!=c)
+    {
+     n=fun(n);
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: What will happen if user wants to calculate factorial of any number greater than 13? Will your code work in that case?

File: factorial/fact.cpp [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    else if (n==c)
+    {
+        std::cout<<"\nXXX Do not enter Character XXX\n";
+         return 0;
+    }
+    else
+    {
+        std::cout<<"\nXXX Do not enter negative Integer XXX\n";
```
Comment (VeerOfficial) [inline factorial/fact.cpp]: This else case will handle everything which is not a positive number or a character. What will happen if user will enter a special character like @/$/& as input?

File: factorial/fact.cpp (hunk 3 above)
Comment (VeerOfficial) [inline factorial/fact.cpp]: Is this case only applicable for n is equal to 0 or should it also be the case for n == 1

Comment (lather10) [reply factorial/fact.cpp]: for n==1 function will give correct output, please review

Comment (VeerOfficial) [reply factorial/fact.cpp]: You resolved this without addressing. Indentation is stil looking weird

**Review by VeerOfficial — APPROVED** [review]

**Pull request merged**